syntax = "proto3";
package sorna;

enum ActionType {
    PING      = 0;
    CREATE    = 1;
    DESTROY   = 2;
    EXECUTE   = 3;
}

message InputMessage {
    ActionType action = 1;
    string kernel_id  = 2;
    int32 length      = 3;
    string content    = 4;
}

enum ReplyType {
    PONG          = 0;
    KERNEL_INFO	  = 1;
}

message OutputMessage {
    ReplyType reply  = 1;
    string kernel_id = 2;
    int32 length     = 3;
    string content   = 4;
}
